atlasdb-tests-shared/src/main/java/com/palantir/timelock/paxos/InMemoryTimelockExtension.java

```diff
@@ -0,0 +1,243 @@
+ */
+
+package com.palantir.timelock.paxos;
+
+import com.google.common.collect.ImmutableSet;
+import com.palantir.atlasdb.factory.TimeLockHelperServices;
+import com.palantir.atlasdb.http.RedirectRetryTargeter;
+import com.palantir.atlasdb.keyvalue.api.LockWatchCachingConfig;
+import com.palantir.atlasdb.keyvalue.api.watch.LockWatchManagerInternal;
+import com.palantir.atlasdb.timelock.AsyncTimelockService;
+import com.palantir.atlasdb.timelock.ConjureTimelockResource;
+import com.palantir.atlasdb.timelock.TimeLockServices;
+import com.palantir.atlasdb.timelock.api.ConjureTimelockService;
+import com.palantir.atlasdb.timelock.lock.LockLog;
+import com.palantir.atlasdb.util.MetricsManager;
+import com.palantir.atlasdb.util.MetricsManagers;
+import com.palantir.conjure.java.api.config.service.PartialServiceConfiguration;
+import com.palantir.conjure.java.api.config.service.UserAgent;
+import com.palantir.conjure.java.serialization.ObjectMappers;
+import com.palantir.lock.LockService;
+import com.palantir.lock.client.CommitTimestampGetter;
+import com.palantir.lock.client.LegacyLeaderTimeGetter;
+import com.palantir.lock.client.LegacyLockTokenUnlocker;
+import com.palantir.lock.client.LockLeaseService;
+import com.palantir.lock.client.NamespacedConjureTimelockServiceImpl;
+import com.palantir.lock.client.RemoteTimelockServiceAdapter;
+import com.palantir.lock.client.RequestBatchersFactory;
+import com.palantir.lock.client.TransactionStarter;
+import com.palantir.lock.v2.NamespacedTimelockRpcClient;
+import com.palantir.lock.v2.TimelockService;
+import com.palantir.logsafe.exceptions.SafeRuntimeException;
+import com.palantir.refreshable.Refreshable;
+import com.palantir.sls.versions.OrderableSlsVersion;
+import com.palantir.timelock.config.ClusterInstallConfiguration;
+import com.palantir.timelock.config.ImmutableClusterInstallConfiguration;
+import com.palantir.timelock.config.ImmutableDefaultClusterConfiguration;
+import com.palantir.timelock.config.ImmutableSqlitePaxosPersistenceConfiguration;
+import com.palantir.timelock.config.ImmutableTimeLockRuntimeConfiguration;
+import com.palantir.timelock.config.PaxosInstallConfiguration;
+import com.palantir.timelock.config.PaxosInstallConfiguration.PaxosLeaderMode;
+import com.palantir.timelock.config.TimeLockInstallConfiguration;
+import com.palantir.timelock.config.TimeLockRuntimeConfiguration;
+import com.palantir.timestamp.ManagedTimestampService;
+import com.palantir.timestamp.TimestampManagementService;
+import com.palantir.timestamp.TimestampService;
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.time.Duration;
+import java.util.List;
+import java.util.Optional;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+public final class InMemoryTimelockExtension
```

Comment:
This change should replace both `InMemoryTimeLockRule` and `InMemoryTimelockServices`. Imho, `InMemoryTimeLockRule` was completely unnecessary in the previous implementation. While upgrading, it's best to simplify things as well.